### PR TITLE
EVG-12866: Add the ability to read a subset of a Report

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-12866

This a part of a ticket to add a new evg command that sends perf data directly to cedar. The users will not be expected to have the necessary evergreen data to generate the report and will only send an array of test results in either a json or bson file.

- add a function `LoadTests` that loads an array of `Test` structures and returns a `Report` empty fields except for the `Test` field.

### CODE REVIEW

File: report_read.go [hunk 1]
```diff
@@ -26,30 +26,15 @@ func getUnmarshaler(fn string) unmarshaler {
...
 
-// LoadReport reads the content of the specified file and attempts to
-// create a Report structure based on the content. The file can be in
-// bson, json, or yaml, and LoadReport examines the files' extension
-// to determine the data format. If the bucket API key, secret, or token are
-// not populated, the corresponding environment variables will be used to
-// populated the values.
+// LoadReport reads the content of the specified file and attempts to create a
+// Report structure based on the content. The file can be in bson, json, or
+// yaml, and LoadReport examines the files' extension to determine the data
```
Comment (Kimchelly) [inline report_read.go:31]: file's

File: report_read.go [hunk 2]
```diff
@@ -26,30 +26,15 @@ func getUnmarshaler(fn string) unmarshaler {
...
+// format. If the bucket API key, secret, or token are not populated, the
+// corresponding environment variables will be used to populated the values.
```
Comment (Kimchelly) [inline report_read.go:33]: populate

**Review by Kimchelly — APPROVED** [review]

**Comment by Kimchelly** [comment]
LGTM mod nits.
